ARCs/arc-00xx.md

```diff
@@ -0,0 +1,235 @@
+of royalties or [vesting](https://en.wikipedia.org/wiki/Vesting)).
+
+Enforcing transfer conditions require freezing the asset and transferring it
+through a clawback operation -- which results in a process that is opaque to
+users and wallets and a bad experience for the users.
+
+The Smart ASA defined by this ARC extends the ASA to increase its expressiveness
+and its flexibility. Developers can now introduce additional business logic
+around its operations. Wallets, dApps, and SDKs can recognize Smart ASAs and
+adjust their flows and user experiences accordingly.
+
+## Specification
+
+The keywords "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be
+interpreted as described in RFC 2119.
+
+The following sections describe:
+
+- The ABI interface for a controlling Smart Contract (the Smart Contract that
+  controls a Smart ASA).
+- The metadata required by a Smart ASA and defining the association between an
+  ASA and its controlling Smart Contract.
+
+### ABI Interface
+
+The ABI interface specified here draws inspiration from the transaction
+[reference](https://developer.algorand.org/docs/get-details/asa/#asset-functions)
+of an Algorand Standard Asset (ASA).
+
+To provide a unified and familiar interface between the Algorand Standard Asset
+and the Smart ASA, method names and parameters have been adapted to the ABI
+types but left otherwise unchanged.
+
+#### Asset Creation
+
+```json
+{
+  "name": "AssetCreate",
+  "args": [
+    { "type": "uint64", "name": "Total" },
+    { "type": "uint32", "name": "Decimals" },
+    { "type": "bool", "name": "DefaultFrozen" },
+    { "type": "string", "name": "UnitName" },
+    { "type": "string", "name": "AssetName" },
+    { "type": "string", "name": "URL" },
+    { "type": "[]byte", "name": "MetaDataHash" },
+    { "type": "address", "name": "ManagerAddr" },
+    { "type": "address", "name": "ReserveAddr" },
+    { "type": "address", "name": "FreezeAddr" },
+    { "type": "address", "name": "ClawbackAddr" }
+  ],
+  "returns": { "type": "uint64" }
+}
+```
+
+Calling `AssetCreate` creates a new Smart ASA and returns the identifier of the
+ASA. The [metadata section](#metadata) describes its required properties.
+
+#### Asset Configuration
+
+```json
+{
+  "name": "AssetConfig",
+  "args": [
+    { "type": "asset", "name": "ConfigAsset" },
+    { "type": "uint64", "name": "Total" },
+    { "type": "uint32", "name": "Decimals" },
+    { "type": "bool", "name": "DefaultFrozen" },
+    { "type": "string", "name": "UnitName" },
+    { "type": "string", "name": "AssetName" },
+    { "type": "string", "name": "URL" },
+    { "type": "[]byte", "name": "MetaDataHash" },
+    { "type": "address", "name": "ManagerAddr" },
+    { "type": "address", "name": "ReserveAddr" },
+    { "type": "address", "name": "FreezeAddr" },
+    { "type": "address", "name": "ClawbackAddr" }
```

Comment:
```suggestion
```

Maybe we should remove the `ClawbackAddr` both from `AssetCreate` and `AssetConfig` since the interface mandates that this can not be different from the `ApplicationAddress`.

Reply:
This is the `ClawbackAddr` of the Smart ASA, not of the underlying ASA (meaning I want to give you access to clawback _through_ the SC).

Reply:
Right! My bad!